napari/_qt/dialogs/qt_notification.py

```diff
@@ -150,9 +137,10 @@ def show(self):
         super().show()
         self.slide_in()
         self.timer = QTimer()
-        self.timer.setInterval(self.DISMISS_AFTER)
-        self.timer.setSingleShot(True)
-        self.timer.timeout.connect(self.close)
+        if self.DISMISS_AFTER > 0:
+            self.timer.setInterval(self.DISMISS_AFTER)
+            self.timer.setSingleShot(True)
+            self.timer.timeout.connect(self.close)
```

Comment:
my thought would be that this would help during testing by not dismissing the notifications; but I can't seem to be able to get a handle on the notifications during the tests; while the same code works in the embeded console.

Reply:
No idea here... Maybe @ziyangczi or @tlambert03 have some ideas?

Reply:
I updated the tests. `test_notification_manager` make sure that the notification manager is capable of intercepting exceptions and warnings, and that it emits events of the correct type... so all we really need to do here is make sure that `NapariQtNotification` is capable of receiving and showing `Notification`, which is what `test_notification_display` does